src/test/java/com/google/sps/servlets/StudyServletTest.java

```diff
@@ -0,0 +1,291 @@
+package com.google.sps.servlets;
+
+import static com.google.sps.tool.Tool.compareJson;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.joda.time.DateTimeUtils;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.KeyFactorySpi;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.sps.data.Card;
+import java.util.List;
+import java.util.ArrayList;
+import java.lang.Math;
+
+@RunWith(JUnit4.class)
+public final class StudyServletTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig()
+                  .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+              new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true)
+          .setEnvIsLoggedIn(true)
+          .setEnvEmail("test@gmail.com")
+          .setEnvAuthDomain("gmail.com");
+
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private StudyServlet servlet;
+  private DatastoreService datastore;
+  private Card cardA, cardB, cardC;
+
+  @Before
+  public void setUp() throws Exception {
+    // Freezes time for testing at 45 hours from january first
+    DateTimeUtils.setCurrentMillisFixed(162000000L);
```

Comment:
Is there something special about this value?

To clarify that it's 45 hours after the epoch you could do 

```java
static long HOURS_MILLIS = 60 * 60 * 1000;

...
DateTimeUtils.setCurrentMillisFixed(45 * HOURS_MILLIS);
```

Reply:
Nothing special I just chose that time timeframe for my tests.

Reply:
done.